src/Markup/Avalonia.Markup/Markup/Parsers/PropertyPathGrammar.cs

```diff
@@ -202,12 +207,26 @@ public override bool Equals(object obj)
                    && other.Name == Name
                    && other.TypeName == TypeName
                    && other.TypeNamespace == TypeNamespace;
+
+            public override int GetHashCode()
+            {
+                int hashCode = 30698940;
+                hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(Name);
+                hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(TypeName);
+                hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(TypeNamespace);
+                return hashCode;
+            }
         }
         
         public class ChildTraversalSyntax : ISyntax
         {
             public static ChildTraversalSyntax Instance { get;  } = new ChildTraversalSyntax();
             public override bool Equals(object obj) => obj is ChildTraversalSyntax;
+
+            public override int GetHashCode()
+            {
+                return base.GetHashCode();
```

Comment:
Hi, what's the point of this change, just to suppress the warning?

I just don't understand such changes, since they fix the consequences, but not the reason